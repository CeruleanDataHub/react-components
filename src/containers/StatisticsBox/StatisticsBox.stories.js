import React from "react";

import { Cell, Grid } from "../Grid/Grid";
import { StatisticsBox } from "./StatisticsBox";

export default {
  title: "StatisticsBox",
  component: StatisticsBox
};

export const StatisticsBoxStory = () => (
  <StatisticsBox title="Value" value={0} growth={0} />
);

export const StatisticsBoxPositiveGrowthStory = () => (
  <StatisticsBox title="Logins / week" value={3789} growth={0.145} />
);

export const StatisticsBoxNegativeGrowthStory = () => (
  <StatisticsBox title="Logins / week" value={3789} growth={-0.145} />
);

export const StatisticsBoxZeroGrowthStory = () => (
  <StatisticsBox title="Logins / week" value={3789} growth={0} />
);

export const StatisticsBoxGreenValueStory = () => (
  <StatisticsBox title="Logins / week" value={3789} growth={0.145} greenValue />
);

export const StatisticsBoxRedValueStory = () => (
  <StatisticsBox title="Logins / week" value={3789} growth={-0.145} redValue />
);

export const StatisticsBoxGridStory = () => (
  <Grid>
    <Cell>
      <StatisticsBox title="Logins / week" value={3789} growth={-0.145} />
    </Cell>
    <Cell>
      <StatisticsBox title="Logins / month" value={17283} growth={0.87} />
    </Cell>
    <Cell>
      <StatisticsBox
        title="Active users / week"
        value={1054}
        growth={-0.062}
        greenValue
      />
    </Cell>
    <Cell>
      <StatisticsBox
        title="Active users / month"
        value={1870}
        growth={0.953}
        redValue
      />
    </Cell>
  </Grid>
);
