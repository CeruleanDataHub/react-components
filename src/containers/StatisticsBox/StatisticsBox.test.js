import React from "react";
import renderer from "react-test-renderer";

import { StatisticsBox } from "./StatisticsBox";

describe("StatisticsBox", () => {
  it("should render", () => {
    const component = renderer.create(
      <StatisticsBox title="Title" value={1} growth={0.3} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render green", () => {
    const component = renderer.create(
      <StatisticsBox title="Title" value={1} growth={0.3} greenValue />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render red", () => {
    const component = renderer.create(
      <StatisticsBox title="Title" value={1} growth={0.3} redValue />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render zero growth", () => {
    const component = renderer.create(
      <StatisticsBox title="Title" value={1} growth={0} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render positive growth", () => {
    const component = renderer.create(
      <StatisticsBox title="Title" value={1} growth={0.3} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render negative growth", () => {
    const component = renderer.create(
      <StatisticsBox title="Title" value={1} growth={-0.3} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should format large values", () => {
    const component = renderer.create(
      <StatisticsBox title="Title" value={10000000000} growth={0.3} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render two decimal points in growth", () => {
    const component = renderer.create(
      <StatisticsBox title="Title" value={1} growth={0.33333333} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
