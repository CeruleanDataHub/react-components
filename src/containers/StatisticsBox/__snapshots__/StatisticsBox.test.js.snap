// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsBox should format large values 1`] = `
<div
  className="sc-AxirZ dGOzPI"
>
  Title
  <div
    className="Grid__Grid-gDLjRO jcWKgV"
  >
    <div
      className="Cell__Cell-tAjrW cZXUpN"
    >
      <span
        className="sc-AxhCb dtGcaw sc-AxiKw fPcOrh"
      >
        10,000,000,000
      </span>
    </div>
    <span
      className="sc-AxhUy mCTBF Cell__Cell-tAjrW hmKLft"
    >
      <span
        className="sc-AxgMl ouThN"
      >
        <span
          className="sc-AxiKw fpvjFK"
        >
          <div
            className="sc-AxheI DfsDh sc-AxjAm jnaJRY lni lni-pyramids"
          />
        </span>
        30%
      </span>
    </span>
  </div>
</div>
`;

exports[`StatisticsBox should render 1`] = `
<div
  className="sc-AxirZ dGOzPI"
>
  Title
  <div
    className="Grid__Grid-gDLjRO jcWKgV"
  >
    <div
      className="Cell__Cell-tAjrW cZXUpN"
    >
      <span
        className="sc-AxhCb dtGcaw sc-AxiKw fPcOrh"
      >
        1
      </span>
    </div>
    <span
      className="sc-AxhUy mCTBF Cell__Cell-tAjrW hmKLft"
    >
      <span
        className="sc-AxgMl ouThN"
      >
        <span
          className="sc-AxiKw fpvjFK"
        >
          <div
            className="sc-AxheI DfsDh sc-AxjAm jnaJRY lni lni-pyramids"
          />
        </span>
        30%
      </span>
    </span>
  </div>
</div>
`;

exports[`StatisticsBox should render green 1`] = `
<div
  className="sc-AxirZ dGOzPI"
>
  Title
  <div
    className="Grid__Grid-gDLjRO jcWKgV"
  >
    <div
      className="Cell__Cell-tAjrW cZXUpN"
    >
      <span
        className="sc-AxhCb dtGcaw sc-AxiKw fPcOrh"
      >
        1
      </span>
    </div>
    <span
      className="sc-AxhUy mCTBF Cell__Cell-tAjrW hmKLft"
    >
      <span
        className="sc-AxgMl ouThN"
      >
        <span
          className="sc-AxiKw fpvjFK"
        >
          <div
            className="sc-AxheI DfsDh sc-AxjAm jnaJRY lni lni-pyramids"
          />
        </span>
        30%
      </span>
    </span>
  </div>
</div>
`;

exports[`StatisticsBox should render negative growth 1`] = `
<div
  className="sc-AxirZ dGOzPI"
>
  Title
  <div
    className="Grid__Grid-gDLjRO jcWKgV"
  >
    <div
      className="Cell__Cell-tAjrW cZXUpN"
    >
      <span
        className="sc-AxhCb dtGcaw sc-AxiKw fPcOrh"
      >
        1
      </span>
    </div>
    <span
      className="sc-AxhUy mCTBF Cell__Cell-tAjrW hmKLft"
    >
      <span
        className="sc-AxgMl ouThN"
      >
        <span
          className="sc-AxiKw gFQOLe"
        >
          <div
            className="sc-AxheI DfsDh sc-AxjAm jnaJRY lni lni-pyramids"
          />
        </span>
        -30%
      </span>
    </span>
  </div>
</div>
`;

exports[`StatisticsBox should render positive growth 1`] = `
<div
  className="sc-AxirZ dGOzPI"
>
  Title
  <div
    className="Grid__Grid-gDLjRO jcWKgV"
  >
    <div
      className="Cell__Cell-tAjrW cZXUpN"
    >
      <span
        className="sc-AxhCb dtGcaw sc-AxiKw fPcOrh"
      >
        1
      </span>
    </div>
    <span
      className="sc-AxhUy mCTBF Cell__Cell-tAjrW hmKLft"
    >
      <span
        className="sc-AxgMl ouThN"
      >
        <span
          className="sc-AxiKw fpvjFK"
        >
          <div
            className="sc-AxheI DfsDh sc-AxjAm jnaJRY lni lni-pyramids"
          />
        </span>
        30%
      </span>
    </span>
  </div>
</div>
`;

exports[`StatisticsBox should render red 1`] = `
<div
  className="sc-AxirZ dGOzPI"
>
  Title
  <div
    className="Grid__Grid-gDLjRO jcWKgV"
  >
    <div
      className="Cell__Cell-tAjrW cZXUpN"
    >
      <span
        className="sc-AxhCb dtGcaw sc-AxiKw fPcOrh"
      >
        1
      </span>
    </div>
    <span
      className="sc-AxhUy mCTBF Cell__Cell-tAjrW hmKLft"
    >
      <span
        className="sc-AxgMl ouThN"
      >
        <span
          className="sc-AxiKw fpvjFK"
        >
          <div
            className="sc-AxheI DfsDh sc-AxjAm jnaJRY lni lni-pyramids"
          />
        </span>
        30%
      </span>
    </span>
  </div>
</div>
`;

exports[`StatisticsBox should render two decimal points in growth 1`] = `
<div
  className="sc-AxirZ dGOzPI"
>
  Title
  <div
    className="Grid__Grid-gDLjRO jcWKgV"
  >
    <div
      className="Cell__Cell-tAjrW cZXUpN"
    >
      <span
        className="sc-AxhCb dtGcaw sc-AxiKw fPcOrh"
      >
        1
      </span>
    </div>
    <span
      className="sc-AxhUy mCTBF Cell__Cell-tAjrW hmKLft"
    >
      <span
        className="sc-AxgMl ouThN"
      >
        <span
          className="sc-AxiKw fpvjFK"
        >
          <div
            className="sc-AxheI DfsDh sc-AxjAm jnaJRY lni lni-pyramids"
          />
        </span>
        33.33%
      </span>
    </span>
  </div>
</div>
`;

exports[`StatisticsBox should render zero growth 1`] = `
<div
  className="sc-AxirZ dGOzPI"
>
  Title
  <div
    className="Grid__Grid-gDLjRO jcWKgV"
  >
    <div
      className="Cell__Cell-tAjrW cZXUpN"
    >
      <span
        className="sc-AxhCb dtGcaw sc-AxiKw fPcOrh"
      >
        1
      </span>
    </div>
    <span
      className="sc-AxhUy mCTBF Cell__Cell-tAjrW hmKLft"
    >
      <span
        className="sc-AxgMl ouThN"
      >
        <span
          className="sc-AxiKw creHPu"
        >
          <div
            className="sc-AxheI DfsDh sc-AxjAm jnaJRY lni lni-pyramids"
          />
        </span>
        0%
      </span>
    </span>
  </div>
</div>
`;
